lib/Sema/TypeCheckType.cpp

```diff
@@ -3980,7 +3980,8 @@ TypeResolver::resolveExistentialType(ExistentialTypeRepr *repr,
                                      TypeResolutionOptions options) {
   auto constraintType = resolveType(repr->getConstraint(),
       options.withContext(TypeResolverContext::ExistentialConstraint));
-  if (constraintType->is<ExistentialMetatypeType>())
+  if (constraintType->is<ExistentialMetatypeType>() ||
+      constraintType->is<ExistentialType>())
```

Comment:
Hmm I don't think this is the correct fix. Right now, `ExistentialMetatype` is special because it really should be `ExistentialType`, I just haven't done that refactoring yet. In general, it is not  correct to write `any` in front of an `any` type. I think your test case below would compile if you assigned a type that does conform to `IteratorProtocol`, but instead it should report an error with a fix-it to remove `any`.

Reply:
Thank you for the explanation!

Reply:
Ah it does make sense, thanks @hborla 
So every `any ProtocolType` is represented as an `ExistentialType`? In this case the type aliasing for an any type is the same as `any any IteratorProtocol` and seems like I miss that. Will make the changes =]

Reply:
Yes, that's correct. Thank you!!